### PR TITLE
json-rpc: Change Public Passphrase Command 

### LINKED ISSUE #1429: Change Public Passphrase
I would like to request the ability to change the public passphrase of wallet.db. Right now, the only command available is dcrctl --wallet walletpassphrasechange "old" "new" but this leaves the public pass unaffected.

### PR DESCRIPTION
**Pull request by Kifen**
This adds a feature to change the public passphrase over json-rpc
`dcrctl --wallet walletpubpassphrasechange <old passphrase> <new passphrase>` 
Resolves #1429

### CODE REVIEW

File: internal/rpchelp/methods.go [hunk 1]
```diff
@@ -102,6 +102,7 @@ var Methods = []struct {
 	{"walletislocked", returnsBool},
 	{"walletlock", nil},
 	{"walletpassphrasechange", nil},
+	{"changepublicpassphrase", nil},
```
Comment (jrick) [inline internal/rpchelp/methods.go]: keep this sorted alphabetically

File: rpc/jsonrpc/methods.go [hunk 2]
```diff
@@ -128,6 +128,7 @@ var handlers = map[string]handler{
 	"walletlock":              {fn: (*Server).walletLock},
 	"walletpassphrase":        {fn: (*Server).walletPassphrase},
 	"walletpassphrasechange":  {fn: (*Server).walletPassphraseChange},
+	"changepublicpassphrase":  {fn: (*Server).changePublicPassphrase},
```
Comment (jrick) [inline rpc/jsonrpc/methods.go]: alphabetical

File: rpc/jsonrpc/methods.go [hunk 3]
```diff
@@ -3527,6 +3528,26 @@ func (s *Server) walletPassphraseChange(ctx context.Context, icmd interface{}) (
...
+func (s *Server) changePublicPassphrase(ctx context.Context, icmd interface{}) (interface{}, error) {
+	cmd := icmd.(*types.WalletPassphraseChangeCmd)
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, errUnloadedWallet
+	}
+
+	err := w.ChangePublicPassphrase([]byte(cmd.OldPassphrase),
+		[]byte(cmd.NewPassphrase))
+	if err != nil {
```
Comment (jrick) [inline rpc/jsonrpc/methods.go]: all of this error handling can be simplified to just:

```
if errors.Is(errors.Passphrase, err) {
    return nil, rpcErrorf(...)
}
return nil, err
```

File: internal/rpchelp/methods.go (hunk 1 above)
Comment (Kifen) [reply internal/rpchelp/methods.go]: It's been renamed...

File: errors/go.mod [hunk 4]
```diff
@@ -1 +1,3 @@
 module github.com/decred/dcrwallet/errors
+
+go 1.12
```
Comment (jrick) [inline errors/go.mod]: Can you remove these changes from the unrelated modules?  PRs should be focused on only changing one thing at a time.

Comment (Kifen) [reply errors/go.mod]: The changes have been removed from the unrelated modules, and CI successful...

**Pull request closed without merging**